contractcourt/htlc_outgoing_contest_resolver.go

```diff
@@ -124,12 +164,18 @@ func (h *htlcOutgoingContestResolver) Resolve() (ContractResolver, error) {
 			// finalized` will be returned and the broadcast will
 			// fail.
 			newHeight := uint32(newBlock.Height)
-			if newHeight >= h.htlcResolution.Expiry {
-				log.Infof("%T(%v): HTLC has expired "+
+			expiry := h.htlcResolution.Expiry
+			if h.isZeroFeeOutput() {
+				expiry--
```

Comment:
?

Reply:
forgot why I did this🤦🏻, removed

Reply:
actually this is needed to make sure the sweeper gets the input before the expiry height is hit, moved the change into a new commit to make it more clear.